libraries/src/Component/ComponentHelper.php

```diff
@@ -217,31 +220,32 @@ public static function filterText($text)
 						$customListAttributes = array_merge($customListAttributes, $tempAttributes);
 					}
 				}
-				elseif ($filterType === 'WL')
+				// "WL" is deprecated in Joomla! 4, will be removed in Joomla! 5
+				elseif (in_array($filterType, ['WL', 'AL']))
 				{
-					$whiteList           = true;
-					$whiteListTags       = array_merge($whiteListTags, $tempTags);
-					$whiteListAttributes = array_merge($whiteListAttributes, $tempAttributes);
+					$allowList       = true;
+					$allowedTags       = array_merge($allowedTags, $tempTags);
+					$allowedAttributes = array_merge($allowedAttributes, $tempAttributes);
 				}
 			}
 		}
 
-		// Remove duplicates before processing (because the blacklist uses both sets of arrays).
-		$blackListTags        = array_unique($blackListTags);
-		$blackListAttributes  = array_unique($blackListAttributes);
+		// Remove duplicates before processing (because the disallowlist uses both sets of arrays).
+		$disallowedTags        = array_unique($disallowedTags);
+		$disallowedAttributes  = array_unique($disallowedAttributes);
 		$customListTags       = array_unique($customListTags);
 		$customListAttributes = array_unique($customListAttributes);
-		$whiteListTags        = array_unique($whiteListTags);
-		$whiteListAttributes  = array_unique($whiteListAttributes);
+		$allowedTags        = array_unique($allowedTags);
+		$allowedAttributes  = array_unique($allowedAttributes);
 
 		if (!$unfiltered)
 		{
-			// Custom blacklist precedes Default blacklist
+			// Custom disallowlist precedes Default disallowlist
 			if ($customList)
 			{
 				$filter = InputFilter::getInstance(array(), array(), 1, 1);
 
-				// Override filter's default blacklist tags and attributes
+				// Override filter's default disallowlist tags and attributes
```

Comment:
```suggestion
				// Override filter's default Disallow List tags and attributes
```
Formalise the wording to show as separate words.